src/ui/src/main/java/com/amazon/sample/ui/web/UtilityController.java

```diff
@@ -51,4 +69,62 @@ private double monteCarloPi(int iterations) {
     public ResponseEntity<String> status(@PathVariable int code) {
         return ResponseEntity.status(code).body("OK");
     }
+
+    @GetMapping(value = "/headers", produces = MediaType.APPLICATION_JSON_VALUE)
+    public ResponseEntity<Map<String, List<String>>> statusHeaders(@RequestHeader HttpHeaders headers) {
+        return ResponseEntity.ok()
+            .body(headers);
+    }
+
+    @GetMapping("/panic")
+    public ResponseEntity<String> panic() {
+        Thread thread = new Thread(() -> {
+            try {
+                Thread.sleep(500); // Small delay to allow response to be sent
+                SpringApplication.exit(context, () -> 1);
+                System.exit(255);
+            } catch (InterruptedException e) {
+                Thread.currentThread().interrupt();
+            }
+        });
+        thread.start();
+        return ResponseEntity.status(HttpStatus.INTERNAL_SERVER_ERROR).body("Shutting down...");
+    }
+
+    // function /echo that answer the hash provided as POST input
+    @PostMapping("/echo")
+    @ResponseBody
+    public ResponseEntity<String> echo(@RequestBody String body) {
+        return ResponseEntity.ok()
+            .body(body);
+    }
+
+    // function /store what take a POST hash to write to a locally created file
+    @PostMapping("/store")
+    @ResponseBody
+    public ResponseEntity<String> store(@RequestBody String body) {
+        String filename = String.valueOf(Math.abs(body.hashCode())); // ensure positive number
+        try (java.io.FileWriter fileWriter = new java.io.FileWriter("/tmp/" + filename + ".json")) {
+            fileWriter.write(body);
+            return ResponseEntity.ok()
+                .body("{\"hash\": \"" + filename + "\"}");
+        } catch (Exception e) {
+            return ResponseEntity.status(HttpStatus.INTERNAL_SERVER_ERROR)
+                .body("Error writing file: " + e.getMessage());
+        }
+    }
+
+    // function /store/{hash} that read a local hash file
+    @GetMapping("/store/{hash}")
+    @ResponseBody
+    public ResponseEntity<String> read(@PathVariable String hash) {
```

Comment:
@guikcd can we put in protections to make sure this can only open files within `/tmp`? Right now you could probably use something like `../root/whatever` to traverse the filesystem. You'll probably have to get the absolute path to the file it tries to load and verify its in `/tmp`.